### DISCUSSION TITLE
서버의 비지니스 로직 구현 중 정리가 필요한 부분

### DISCUSSION THREAD

**Discussion by gomarag** [codestates/siphoon-note#123] [General]
1. POST, PATCH, DELETE 와 같이 **기존 데이터에 변화를 발생시키는 요청**은 **서버가 변경된 데이터의 고유 아이디를 반환하는 것**으로 해결이 되는지
   - 예) 새로운 글을 쓰는 POST 요청 시, 서버가 추가된 글 객체를 반환하면 바로 화면에 반영이 되는지
   - 예) 기존에 작성한 글을 휴지통으로 보내는 PATCH 요청 후, 서버가 isDeleted 필드가 수정된 글 아이디를 반환하면 출력화면에서 휴지통으로 보내지게 되는지
  
2. `POST /signup` 요청에 대한 응답으로 accessToken과 refreshToken을 발급한다는 것은 **가입 후 바로 로그인까지 완료되어야 하는 것**으로 보임. 그렇지 않으면 쿠키에 담아 반환한 accessToken을 어딘가에 저장해야 하기 때문.

**Comment by gomarag** [comment]
1. @Leekyeongmi  `글 CRUD`에서 기존 데이터의 변경이 생기는 경우 서버가 `변경된 데이터 객체` 나 `변경된 데이터의 고유 아이디`만 반환해주면 괜찮을지 피드백 주시면 좋을 것 같아요

2. @lindist12 @jongyeol12  `회원가입` 성공하면 서버가 accessToken + refreshToken을 발급하면서 해당 유저를 바로 `로그인 상태`로 만들어서 `/essays` 페이지로 보내는게 괜찮은지 피드백 주시면 좋을 것 같아요

**Reply by Leekyeongmi** [reply]
변경된 데이터 객체만 보내주시면 될 거 같습니다!

**Comment by Leekyeongmi** [comment]
[클라이언트 질문/고민]

1) 랜딩페이지 공개글 데이터
이메일, 내용, 작성한 날짜, 태그로 되어 있는데 **유저 닉네임이 아닌 이메일인지? 해당 글의 태그들도 전달되는지?**

2) 다이어리페이지 글 작성
**body에 isPublic 여부도 바디에 함께 전송하는 것이 맞는지?** sample에 나와있지 않아서 여쭤봅니다!

3) 태그와 휴지통 글 조회 
태그와 휴지통 글은 EssayList를 받아온 후에 렌더링이 가능한데, 문제는 EssayList 전체를 받아오는 것이 아니라 20개 단위로 잘라서 무한스크롤으로 가져온다. #110 가람님 디스커션에 보면, 전체 글을 주면 클라이언트에서 필드에 따라 필터링 하면 어떨지 제안으로 되어 있는데 **전체 EssayList를 받는 것도 가능한지?**

[알림]
@lindist12  @jongyeol12 서버에서 응답을 받을 때, first project에서는 res.data.message로 응답을 확인했지만, 이번에는 status라는 속성으로 하기로 했으니 그렇게 변경하는 코드로 짜야 합니다! 준형님 작성하신 코드 수정 추후에 부탁드릴게요!

**Reply by gomarag** [reply]
1. 이메일이 아닌 유저 이름을 전달하려고 합니다. 
- [x] api에 유저 이름을 반환하는 것으로 수정할게요! 태그는 필요에 따라서 전달할 수도 전달하지 않을수도 있습니다.

2. 새 글 작성 시 공개 여부를 isPublic으로 받을 예정입니다. 
- [x] 이 부분 역시 api에 누락된 것 같아 추가하겠습니다!

3. 일전에 제가 디스커션 #110 에서 말씀 드린 'essayList 전체'라는 용어가 모호하게 적혀있어서 정정하겠습니다. 제가 의도했던 바는 isDeleted가 T/F인지와 상관없이 작성했던 총 글 개수가 30개라면, 글 객체 0 ~ 29 인덱스까지의 글 리스트 입니다. 여기에서 '전체'라고 얘기했던 이유는 /trashes와 /essays로 글 객체들을 분리하지 않고 GET /essays 요청 하나에 모두 보내기 때문에 '전체 목록'이라는 단어를 사용한 것 같습니다. 혼동을 드려서 죄송해요.

**Reply by gomarag** [reply]
### 22-04-05-화 미팅 이후 백엔드의 수정 계획
> "수정 완료 후 체크박스에 표시하고 디스코드에 메시지로 남기겠습니다😃"
- Swagger API 수정 사항
    - [x]  `GET /` 응답 객체에 essay.writer 추가
    - [x]  `POST /essays` req.body에 tag.tagName만 받도록 + isPublic 필드 추가
    - [x]  `GET /essays` 에서 보내는 글 목록은 `isDeleted:false` 인 글 객체만 포함 + 작성일 `yyyy-mm-dd`로 다듬어서 보낼 것
    - [x]  `GET /trashes` 휴지통 목록 조회 추가 → 응답: 복구, 영구삭제를 위한 essayId 필수로 포함, 맨 첫줄이나 작성일자를 추출하여 보내려고 함
    - [x]  `GET /tags` 태그 전체 조회 추가
   
- client에서 사용하는 API 요청 config 수정 사항
    - [x]  호스트 새 주소로 수정
    - [x]  추가된 기능 업데이트
    - [x]  사용법 주석
    - [x]  오타 수정

**Reply by Leekyeongmi** [reply]
### 22-04-06 추가 질문/논의 내용
- [x] 랭킹에서 받는 프로필 이미지는 url 주소가 아니라 `0` `1` 등의 string 타입의 0부터 15까지의 랜덤 숫자를 받을 수 있는지?
- [x] 랭킹 유저 리스트는 1등 -> 4등 순서대로 배열에 담겨 오는 것인지?
- [x] 분석 페이지에 들어갈 recordList 값이 배열에 담겨 오는데, 그냥 객체 형태로 와도 괜찮은지?
- [x] 키워드 관련 API 
- [x] 랜딩페이지에서 usageDate가 아닌 작성한 글 개수로 수정

**Reply by gomarag** [reply]
### 22-04-06 해당 논의에 대한 답변
- [x] User 고유의 ProfileImage는 `{ id, url }` 형식의 객체가 아닌 `0 ~ 15` 까지의 정수를 갖도록 수정 완료
- [x] `GET /` 요청에 대한 응답 데이터 중 userList는 totalEssay 값을 기준으로 정렬할 예정
- [x] `GET /essays` 요청에 대한 응답 중 `recordList` 대신 객체 `record`로 수정 완료
- [x] `글감 목록 aka 키워드`는 db에 저장된 30개의 단어 중 해당 일자의 `키워드` 하나를 `GET /essays` 응답에 `todaysWord`로 보내도록 수정 완료. 
- [x]  체험모드에서 사용할 단어 목록은 클라이언트가 관리하도록 협의함
